.github/workflows/license.yml

```diff
@@ -0,0 +1,81 @@
+name: Graph Compiler License Check
+
+on:
+  pull_request:
+
+permissions: read-all
+
+jobs:
+  license-check:
+    name: License Check
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      repository-projects: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Get merge base
+        run: |
+          echo "MERGE_BASE=`git merge-base ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}`" >> $GITHUB_ENV
+
+      - name: Get changed files
+        run: |
+          echo "CHANGED_FILES=`git diff --name-only $MERGE_BASE ${{ github.event.pull_request.head.sha }} | paste -sd,`" >> $GITHUB_ENV
+
+      - name: Perform license check
+        shell: python
+        run: |
+            import os, datetime, sys
+
+            intel_license: list[str] = [
+                'Copyright (C) YYYY Intel Corporation',
+                'Licensed under the Apache License, Version 2.0 (the "License");',
+                'you may not use this file except in compliance with the License.',
+                'You may obtain a copy of the License at',
+                'http://www.apache.org/licenses/LICENSE-2.0',
+                'Unless required by applicable law or agreed to in writing,',
+                'software distributed under the License is distributed on an "AS IS" BASIS,',
+                'WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.',
+                'See the License for the specific language governing permissions',
+                'and limitations under the License.',
+                'SPDX-License-Identifier: Apache-2.0',
+            ]
+
+            llvm_license: list[str] = [
+                'This file is licensed under the Apache License v2.0 with LLVM Exceptions.',
```

Comment:
LLVM's header looks like

```
//===- Example.cpp - Tests for Example ------------------------------------===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//
```

Shall we check the header `//===- Example.cpp - Tests for Example ------------------------------------===//` and the footer `//===----------------------------------------------------------------------===//` as well?

Note that the header has variable number of `------------------------------------` to align the line wrap limit.

Reply:
@Menooker 
The string `Tests for Example` you mentioned is not a costant string or from the filename. I do not know how to verify it.
That's why I skipped the first line check for llvm license.

Reply:
It is ` {file name} - {desc} ---------`. I think we can check the filename and the number of `---------`. For auto-fix, we can fill-in the filename and a place holder "{DESC}" and auto fix the number of `------------`.

In checking mode, we can throw an error when detecting "{DESC}".

A workflow for the user to use auto-fix should be:

1. write code without LLVM header
2. auto-fix for the first time
3. fill in {DESC} in the code
4. auto-fix again, maybe check if there are remaining {DESC}